ethstorage/storage_manager.go

```diff
@@ -293,6 +302,143 @@ func (s *StorageManager) syncCheck(kvIdx uint64) error {
 	return nil
 }
 
+// DownloadAllMetas This function download the blob hashes of all the local storage shards from the smart contract
+func (s *StorageManager) DownloadAllMetas() error {
+	s.mu.Lock()
```

Comment:
This will lock other threads that accessing mu.Lock() - since the call of this method can be costly, it is intentional?